### PR TITLE
Update `react-intl` to 5.20.2

### PR DESCRIPTION
**Pull request by SudharakaP**
Related to https://github.com/opencollective/opencollective-frontend/pull/6471

This fixes what's mentioned in https://github.com/opencollective/opencollective-frontend/pull/6471#issuecomment-863230656 🐨 

image

before this PR we had the following problem with the tier cards for example,

edit collective -- edit tiers (failed)

### CODE REVIEW

**Review by znarf — APPROVED** [review]
Thank you, looks good.

Was the change documented in `react-intl`?

cc @Betree so you know about the change.

**Review by Betree — COMMENTED** [review]
Interesting, could that fix https://github.com/opencollective/opencollective/issues/3095 as well?

I see multiple strings not using `other` with a select, are they not impacted too? You can search using this regex `, select((?!other).)+$` in `en.json`:

```
  92,57:   "AmountInterval": "{amount} {currencyCode} / {interval, select, month {mo.} year {yr.}}",
  93,61:   "AmountIntervalLong": "{amount} {currencyCode} / {interval, select, month {month} year {year}}",
  297,59:   "collective.event.delete.isNotDeletable-message": "{type, select, EVENT {Events} PROJECT {Projects}} with transactions, orders or paid expenses cannot be deleted. Please archive it instead.",
  525,187:   "ContributionSummary.NextCharge": "If you select PayPal, you will be charged on the same day each month. Otherwise the next charge will be on {date} and then the first day of {interval, select, month {each month} year {the same month each year}}. You can cancel or edit this contribution by going to 'Manage Contributions'.",
  705,82:   "editCollective.host.balance": "It currently holds {balance} on behalf of {type, select, COLLECTIVE {your Collective} FUND {your Fund}}.",
  707,107:   "editCollective.host.change.balanceNotEmpty": "To change your Fiscal Host, you first need to empty {type, select, COLLECTIVE {your Collective's balance} FUND {your Fund's balance}}. You can do this by submitting expenses, making financial contributions, or sending the balance to your Fiscal Host using the {emptyBalanceLink} feature.",
  733,111:   "editCollective.selfHost.change.balanceNotEmpty": "To change your Fiscal Host, you first need to empty {type, select, COLLECTIVE {your Collective's balance} FUND {your Fund's balance}} by submitting and paying expenses.",
  734,42:   "editCollective.selfHost.label": "{type, select, COLLECTIVE {Your Collective} FUND {Your Fund}} hold its own funds; it doesn't use a Fiscal Host.",
  1354,35:   "membership.description": "{role, select, ADMIN {Admin} MEMBER {Core Contributor} BACKER {Contributor}} since {date}",
  1566,37:   "paymentMethod.typeSelect": "{type, select, giftcard {Gift card} creditcard {Credit card} prepaid {Prepaid}}",
```

**Comment by SudharakaP** [comment]
> Thank you, looks good.
> 
> Was the change documented in `react-intl`?
> 
> cc @Betree so you know about the change.

@znarf : Unfortunately no, it wasn't documented and I searched a lot and couldn't find anything in that repo about this. I arrived at this solution quite by chance reading some stuff at the [ICU specification](https://unicode-org.github.io/icu/userguide/icu_data/). Tomorrow I'll open an issue in that repo for this, to see if anyone else had this issue as well. 🤔 😄 

> Interesting, could that fix [opencollective/opencollective#3095](https://github.com/opencollective/opencollective/issues/3095) as well?
> 
> I see multiple strings not using `other` with a select, are they not impacted too? You can search using this regex `, select((?!other).)+$` in `en.json`:
> 
> ```
>   92,57:   "AmountInterval": "{amount} {currencyCode} / {interval, select, month {mo.} year {yr.}}",
>   93,61:   "AmountIntervalLong": "{amount} {currencyCode} / {interval, select, month {month} year {year}}",
>   297,59:   "collective.event.delete.isNotDeletable-message": "{type, select, EVENT {Events} PROJECT {Projects}} with transactions, orders or paid expenses cannot be deleted. Please archive it instead.",
>   525,187:   "ContributionSummary.NextCharge": "If you select PayPal, you will be charged on the same day each month. Otherwise the next charge will be on {date} and then the first day of {interval, select, month {each month} year {the same month each year}}. You can cancel or edit this contribution by going to 'Manage Contributions'.",
>   705,82:   "editCollective.host.balance": "It currently holds {balance} on behalf of {type, select, COLLECTIVE {your Collective} FUND {your Fund}}.",
>   707,107:   "editCollective.host.change.balanceNotEmpty": "To change your Fiscal Host, you first need to empty {type, select, COLLECTIVE {your Collective's balance} FUND {your Fund's balance}}. You can do this by submitting expenses, making financial contributions, or sending the balance to your Fiscal Host using the {emptyBalanceLink} feature.",
>   733,111:   "editCollective.selfHost.change.balanceNotEmpty": "To change your Fiscal Host, you first need to empty {type, select, COLLECTIVE {your Collective's balance} FUND {your Fund's balance}} by submitting and paying expenses.",
>   734,42:   "editCollective.selfHost.label": "{type, select, COLLECTIVE {Your Collective} FUND {Your Fund}} hold its own funds; it doesn't use a Fiscal Host.",
>   1354,35:   "membership.description": "{role, select, ADMIN {Admin} MEMBER {Core Contributor} BACKER {Contributor}} since {date}",
>   1566,37:   "paymentMethod.typeSelect": "{type, select, giftcard {Gift card} creditcard {Credit card} prepaid {Prepaid}}",
> ```

@Betree : Thanks a bunch for the regex. I think I missed these. Will update it first thing tomorrow. 🙏🏽 I am sure the issue you mentioned also is similar to this. I'll investigate that one as well. 😄

**Comment by SudharakaP** [comment]
@Betree : I've adjusted these strings at all the other places as well. Let me know if there's anything else to be done here. 😄 

Also I've opened a new issue in the `react-intl` repo just to see if this issue is something intended or not on the `react-intl` side; https://github.com/formatjs/formatjs/issues/2987

File: lang/en.json [hunk 1]
```diff
@@ -294,7 +294,7 @@
   "collective.editHost.withdrawApp": "Are you sure you want to withdraw your application to {name}?",
   "collective.emptyBalance.body": "Are you sure you want to {action} to {collective}?",
   "collective.emptyBalance.header": "{action} Balance",
-  "collective.event.delete.isNotDeletable-message": "{type, select, EVENT {Events} PROJECT {Projects}} with transactions, orders or paid expenses cannot be deleted. Please archive it instead.",
+  "collective.event.delete.isNotDeletable-message": "{type, select, EVENT {Events} PROJECT {Projects} other {}} with transactions, orders or paid expenses cannot be deleted. Please archive it instead.",
```
Comment (Betree) [inline lang/en.json:297]: A default string for this one could actually be nice

```suggestion
  "collective.event.delete.isNotDeletable-message": "{type, select, EVENT {Events} PROJECT {Projects} other {Accounts}} with transactions, orders or paid expenses cannot be deleted. Please archive it instead.",
```

File: lang/en.json [hunk 2]
```diff
@@ -563,7 +563,7 @@
   "CP.Contributions.ConnectedCollective": "{n, plural, one {This Collective is} other {These Collectives are}} connected to us",
   "CP.Contributions.PartOfOrg": "{n, plural, one {This Collective is} other {These Collectives are}} part of our Organization",
   "create.collective.placeholder": "Making the world a better place",
-  "create.pledge.stats": "by {orgCount, plural, =0 {} one {# sponsor} other {# sponsors}} {both, plural, =0 {} one { and }} {userCount, plural, =0 {} one {# backer } other {# backers }}",
+  "create.pledge.stats": "by {orgCount, plural, =0 {} one {# sponsor} other {# sponsors}} {both, plural, =0 {} one { and } other {}} {userCount, plural, =0 {} one {# backer } other {# backers }}",
```
Comment (Betree) [inline lang/en.json:566]: Maybe we can simplify? (not sure if this works)

```suggestion
  "create.pledge.stats": "by {orgCount, plural, =0 {} one {# sponsor} other {# sponsors}} {both, plural, =0 {} other { and }} {userCount, plural, =0 {} one {# backer } other {# backers }}",
```

File: lang/en.json [hunk 3]
```diff
@@ -702,9 +702,9 @@
   "Edit": "Edit",
   "editCollective.fiscalHosting": "Fiscal Hosting",
   "EditCollective.GSTNumber": "GST number",
-  "editCollective.host.balance": "It currently holds {balance} on behalf of {type, select, COLLECTIVE {your Collective} FUND {your Fund}}.",
+  "editCollective.host.balance": "It currently holds {balance} on behalf of {type, select, COLLECTIVE {your Collective} FUND {your Fund} other {}}.",
```
Comment (Betree) [inline lang/en.json:705]:
```suggestion
  "editCollective.host.balance": "It currently holds {balance} on behalf of {type, select, COLLECTIVE {your Collective} FUND {your Fund} other {your account}}.",
```

File: lang/en.json [hunk 4]
```diff
@@ -702,9 +702,9 @@
...
   "editCollective.host.cancelApplicationBtn": "Withdraw application",
-  "editCollective.host.change.balanceNotEmpty": "To change your Fiscal Host, you first need to empty {type, select, COLLECTIVE {your Collective's balance} FUND {your Fund's balance}}. You can do this by submitting expenses, making financial contributions, or sending the balance to your Fiscal Host using the {emptyBalanceLink} feature.",
+  "editCollective.host.change.balanceNotEmpty": "To change your Fiscal Host, you first need to empty {type, select, COLLECTIVE {your Collective's balance} FUND {your Fund's balance} other {}}. You can do this by submitting expenses, making financial contributions, or sending the balance to your Fiscal Host using the {emptyBalanceLink} feature.",
```
Comment (Betree) [inline lang/en.json:707]:
```suggestion
  "editCollective.host.change.balanceNotEmpty": "To change your Fiscal Host, you first need to empty {type, select, COLLECTIVE {your Collective's balance} FUND {your Fund's balance} other {your balance}}. You can do this by submitting expenses, making financial contributions, or sending the balance to your Fiscal Host using the {emptyBalanceLink} feature.",
```

File: lang/en.json [hunk 5]
```diff
@@ -730,8 +730,8 @@
   "editCollective.rollingLimit.enable": "Enable 2FA for payouts",
   "editCollective.rollingLimit.label": "Rolling payout limit",
   "editCollective.selfHost.balance": "Current balance: {balance}.",
-  "editCollective.selfHost.change.balanceNotEmpty": "To change your Fiscal Host, you first need to empty {type, select, COLLECTIVE {your Collective's balance} FUND {your Fund's balance}} by submitting and paying expenses.",
-  "editCollective.selfHost.label": "{type, select, COLLECTIVE {Your Collective} FUND {Your Fund}} hold its own funds; it doesn't use a Fiscal Host.",
+  "editCollective.selfHost.change.balanceNotEmpty": "To change your Fiscal Host, you first need to empty {type, select, COLLECTIVE {your Collective's balance} FUND {your Fund's balance} other {}} by submitting and paying expenses.",
```
Comment (Betree) [inline lang/en.json:733]:
```suggestion
  "editCollective.selfHost.change.balanceNotEmpty": "To change your Fiscal Host, you first need to empty {type, select, COLLECTIVE {your Collective's balance} FUND {your Fund's balance} other {your balance}} by submitting and paying expenses.",
```

File: lang/en.json [hunk 6]
```diff
@@ -730,8 +730,8 @@
...
+  "editCollective.selfHost.label": "{type, select, COLLECTIVE {Your Collective} FUND {Your Fund} other {}} hold its own funds; it doesn't use a Fiscal Host.",
```
Comment (Betree) [inline lang/en.json:734]:
```suggestion
  "editCollective.selfHost.label": "{type, select, COLLECTIVE {Your Collective} FUND {Your Fund} other {Your account}} hold its own funds; it doesn't use a Fiscal Host.",
```

File: lang/en.json [hunk 7]
```diff
@@ -1633,7 +1633,7 @@
   "pledge.contactToClaim": "To claim this Collective, contact <SupportLink></SupportLink>.",
   "pledge.definition": "A pledge is a way to show interest in supporting a cause or project that is not yet on Open Collective, just like {collective}. If they create a Collective, you will receive an email asking you to fulfill your pledge.",
   "pledge.ownerQuestion": "Do you own {collective}?",
-  "pledge.stats": "{orgCount, plural, =0 {} one {# organization} other {# organizations}} {both, plural, =0 {} one { and }} {userCount, plural, =0 {} one {# individual } other {# individuals }} {totalCount, plural, one {has } other {have }} already pledged a total of {amount} {currency}",
+  "pledge.stats": "{orgCount, plural, =0 {} one {# organization} other {# organizations}} {both, plural, =0 {} one { and } other {}} {userCount, plural, =0 {} one {# individual } other {# individuals }} {totalCount, plural, one {has } other {have }} already pledged a total of {amount} {currency}",
```
Comment (Betree) [inline lang/en.json:1636]: If it works here also

```suggestion
  "pledge.stats": "{orgCount, plural, =0 {} one {# organization} other {# organizations}} {both, plural, =0 {} other { and }} {userCount, plural, =0 {} one {# individual } other {# individuals }} {totalCount, plural, one {has } other {have }} already pledged a total of {amount} {currency}",
```

Comment (SudharakaP) [reply lang/en.json]: Yes this works. 😄

File: lang/en.json (hunk 2 above)
Comment (SudharakaP) [reply lang/en.json]: Yes, this works. It think the other needs to be there; that's the only condition. 😄

**Comment by SudharakaP** [comment]
@Betree : I've added all your suggestions. Let me know if you have more. 😄

**Review by Betree — APPROVED** [review]
Lgtm! Let me know if you want me to try this on Safari

**Comment by znarf** [comment]
Great stuff!

**Comment by SudharakaP** [comment]
> Great stuff!

Thanks. 😄

**Comment by Betree** [comment]
> I'll take a look at it a bit more. But I'll merge this PR if there's no objections. koala

No rush, the Safari issue is only impacting a small percentage of users. I was just curious to see if it could be related, but we don't need to solve it now :wink:

**Pull request merged**